Koha/Illrequest.pm

```diff
@@ -1346,57 +1330,282 @@ EOF
-                replyto     => $replyto,
-                subject     => Encode::encode( "utf8", $params->{subject} ),
-                message     => Encode::encode( "utf8", $params->{body} ),
-                contenttype => 'text/plain',
+        # So we get a notice hashref, then substitute the possibly
+        # modified title and body from the draft stage
+        my $letter = $self->get_notice({
+            notice_code => 'ILL_PARTNER_REQ',
+            transport   => 'email'
+        });
+        $letter->{title} = $params->{subject};
+        $letter->{content} = $params->{body};
+
+        # Send the email
+        my $params = {
+            letter                 => $letter,
+            borrowernumber         => $self->borrowernumber,
+            message_transport_type => 'email',
+            to_address             => $to,
+            from_address           => $from
+        };
+
+        if ($letter) {
+            my $result = C4::Letters::EnqueueLetter($params);
+            if ( $result ) {
+                $self->status("GENREQ")->store;
+                $self->_backend_capability(
+                    'set_requested_partners',
+                    {
+                        request => $self,
+                        to => $to
+                    }
+                );
+                return {
+                    error   => 0,
+                    status  => '',
+                    message => '',
+                    method  => 'generic_confirm',
+                    stage   => 'commit',
+                    next    => 'illview',
+                };
             }
-        );
-        # Send it
-        my $result = sendmail(%mail);
-        if ( $result ) {
-            $self->status("GENREQ")->store;
-            $self->_backend_capability(
-                'set_requested_partners',
-                {
-                    request => $self,
-                    to => $to
-                }
-            );
-            return {
-                error   => 0,
-                status  => '',
-                message => '',
-                method  => 'generic_confirm',
-                stage   => 'commit',
-                next    => 'illview',
-            };
-        } else {
-            return {
-                error   => 1,
-                status  => 'email_failed',
-                message => $Mail::Sendmail::error,
-                method  => 'generic_confirm',
-                stage   => 'draft',
-            };
         }
+        return {
+            error   => 1,
+            status  => 'email_failed',
+            message => 'Email queueing failed',
+            method  => 'generic_confirm',
+            stage   => 'draft',
+        };
     } else {
         die "Unknown stage, should not have happened."
     }
 }
 
+=head3 get_staff_to_address
+
+    my $email = $request->get_staff_to_address();
+
+Get the email address to which staff notices should be sent
+
+=cut
+
+sub get_staff_to_address {
```

Comment:
I feel like this should be in the Koha::Library class instead of here...  perhaps beneath the `inbound_email_address` method introduced by bug 22823.

Reply:
This whole method could be simplified by using `inbound_email_address` as a fallback after checking for `branchillemail` and `ILLDefaultStaffEmail`.